include/z64state.h

```diff
@@ -0,0 +1,39 @@
+#ifndef Z64STATE_H
```

Comment:
Are we going with `z64state.h` instead of `z64gamestate.h`? :thinking:

Reply:
I don't think it was a choice on my part, just loosely picked one. I'm fine changing it to z64gamestate.h, probably better anyway given the `GameState_` prefix used in the implementation

Reply:
fwiw i agree with gamestate
we should at some point decide on game state vs gamestate. Ive always tended to call it one word, but the capitalization in things like GameState suggests it should have a space/underscore

Reply:
doesn't help that og headers like bgcheck don't have underscores

Reply:
Another name idea: name this header `z64game.h` because of `z_game.c`
I think this is the name I like the most because it would mean this header is the header for the "game" system

Reply:
Will do eventually
(with the provision as stated on discord that "game" may not be the final name for the system)
https://discord.com/channels/688807550715560050/838852326231769089/1084870531662217236